File: fc-chat/demo/part3/ch3/05-2-message-client-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/CreateResponse.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.ChannelId;

public class CreateResponse extends BaseMessage {

  private final ChannelId channelId;
  private final String title;

  @JsonCreator
  public CreateResponse(
      @JsonProperty("channelId") ChannelId channelId, @JsonProperty("title") String title) {
    super(MessageType.CREATE_RESPONSE);
    this.channelId = channelId;
    this.title = title;
  }

  public ChannelId getChannelId() {
    return channelId;
  }

  public String getTitle() {
    return title;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/outbound/CreateResponse.kt
package com.yscorp.withpush.messagesystem.dto.websocket.outbound

import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.ChannelId


class CreateResponse(val channelId: ChannelId, val title: String) : BaseMessage(MessageType.CREATE_RESPONSE) {

    fun getChannelId(): ChannelId {
        return channelId
    }
}
